packages/ccd-js-gen/src/lib.ts

```diff
@@ -0,0 +1,275 @@
+        });
+
+        classDecl
+            .addConstructor({
+                docs: ['Contruct a client for a contract instance on chain'],
+                parameters: [
+                    {
+                        name: grpcClientId,
+                        type: 'SDK.ConcordiumGRPCClient',
+                        scope: tsm.Scope.Public,
+                    },
+                    {
+                        name: contractAddressId,
+                        type: 'SDK.ContractAddress',
+                        isReadonly: true,
+                        scope: tsm.Scope.Public,
+                    },
+                ],
+            })
+            .setBodyText(
+                `this.${genericContractId} = new SDK.Contract(${grpcClientId}, ${contractAddressId}, '${contract.contractName}');
+this.${dryRunId} = new ${contractDryRunClassId}(this.${genericContractId});`
+            );
+
+        dryRunClassDecl.addConstructor({
+            docs: ['Contruct a client for a contract instance on chain'],
+            parameters: [
+                {
+                    name: genericContractId,
+                    type: 'SDK.Contract',
+                    scope: tsm.Scope.Private,
+                },
+            ],
+        });
+
+        for (const entrypointName of contract.entrypointNames) {
+            const transactionMetadataId = 'transactionMetadata';
+            const parameterId = 'parameter';
+            const signerId = 'signer';
+            classDecl
+                .addMethod({
+                    docs: [
+                        `Send a message to the '${entrypointName}' entrypoint of the '${contract.contractName}' contract.`,
+                    ],
+                    scope: tsm.Scope.Public,
+                    name: toCamelCase(entrypointName),
+                    parameters: [
+                        {
+                            name: transactionMetadataId,
+                            type: 'SDK.ContractTransactionMetadata',
+                        },
+                        {
+                            name: parameterId,
+                            type: 'SDK.HexString',
+                        },
+                        {
+                            name: signerId,
+                            type: 'SDK.AccountSigner',
+                        },
+                    ],
+                    returnType: 'Promise<SDK.HexString>',
+                })
+                .setBodyText(
+                    `return this.${genericContractId}.createAndSendUpdateTransaction(
+    '${entrypointName}',
+    SDK.encodeHexString,
+    ${transactionMetadataId},
+    ${parameterId},
+    ${signerId}
+);`
+                );
+            const blockHashId = 'blockHash';
+            dryRunClassDecl
+                .addMethod({
+                    docs: [
+                        `Dry run a message to the '${entrypointName}' entrypoint of the '${contract.contractName}' contract`,
+                    ],
```

Comment:
Should we add more comprehensive documentation to each method? Would be nice if the result would be in the form of

```javascript
/**
 * Dry run a message to the '<entrypointName>' entrypoint of the '<contract.contractName>' contract
 *
 * @param {SDK.HexString} parameter - Hex encoded parameter for entrypoint
 * @param {SDK.HexString} [blockHash] - Block hash of the block to invoke entrypoint at
 * 
 * @throws If the entrypoint is not successfully invoked.
 * 
 * @returns {SDK.HexString} Hex encoded response
 */
```